src/client.rs

```diff
@@ -20,85 +30,105 @@ impl Client {
         Ok(Client { client, settings })
     }
     /// Sends events to a given stream.
-    pub fn write_events<S>(&self, stream: S) -> commands::WriteEvents
+    pub async fn append_to_stream<StreamName, Events>(
+        &self,
+        stream: StreamName,
+        options: &AppendToStreamOptions,
+        events: Events,
+    ) -> crate::Result<Result<WriteResult, WrongExpectedVersion>>
     where
-        S: AsRef<str>,
+        StreamName: AsRef<str>,
+        Events: ToEvents + 'static,
     {
-        commands::WriteEvents::new(
-            self.client.clone(),
-            stream.as_ref().to_string(),
-            self.settings.default_user_name.clone(),
-        )
+        commands::append_to_stream(&self.client, stream, options, events.to_events()).await
     }
 
     /// Reads events from a given stream. The reading can be done forward and
     /// backward.
-    pub fn read_stream<S>(&self, stream: S) -> commands::ReadStreamEvents
+    pub async fn read_stream<StreamName, Count>(
+        &self,
+        stream: StreamName,
+        options: &ReadStreamOptions,
+        count: Count,
```

Comment:
**SUGGESTION:** Can we rename it to maxCount to be aligned with other clients? See https://github.com/EventStore/EventStore-Client-NodeJS/pull/130.

Reply:
Nope, the encoding I did is better: https://github.com/EventStore/EventStoreDB-Client-Rust/pull/49#discussion_r573647195

Reply:
It simulates function overloading without having function overloading.